baseframe/forms/validators.py

```diff
@@ -161,6 +162,51 @@ def compare(self, value, other):
         return value != other
 
 
+class IsPublicEmailDomain(object):
+    def __init__(self, message=None):
+        self.message = message or _(u'This domain is not a public email domain.')
+
+    def make_cache_key(self, email_or_domain):
+        return 'mxrecord/' + email_or_domain
+
+    def get_mx(self, email_or_domain):
+        cache_key = self.make_cache_key(email_or_domain)
+        mx_cache = asset_cache.get(cache_key)
+
+        if mx_cache and isinstance(mx_cache, dict):
+            domain = mx_cache.get('domain')
+        else:
+            domain = None
+
+        if domain is None:
+            # Something is wrong with the cache, fetch result again and set cache
+            sniffedmx = mxsniff(email_or_domain)
```

Comment:
Check for exceptions here. `mxsniff`'s documentation includes this:

```rst
:raises MXLookupException: If a DNS lookup error happens and ``ignore_errors`` is False
```

If an exception happens, the validator must:

1. Fail in `IsPublicEmailDomain` (since we were unable to confirm that it _is_ a public email domain)
2. Pass in `IsNotPublicEmailDomain` (since our expectation is that most domains are not public email domains)

The documentation for both validators must clearly specify this behaviour.